src/lib/serviceRoutinesV2/semStateTreat.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+*
+* Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string>
+#include <vector>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+#include "common/statistics.h"
+#include "common/clockFunctions.h"
+#include "common/tag.h"
+
+#include "ngsi/ParseData.h"
+#include "rest/ConnectionInfo.h"
+#include "rest/OrionError.h"
+#include "rest/uriParamNames.h"
+#include "serviceRoutinesV2/semStateTreat.h"
+#include "alarmMgr/alarmMgr.h"
+#include "mongoBackend/mongoConnectionPool.h"
+
+
+
+/* ****************************************************************************
+*
+* semRender - 
+*/
+static const std::string semRender(const char* name, bool toplevel, const char* state)
```

Comment:
In that case, I'd suggest to add a comment about it in the function preamble, eg.:

"Note that toplevel is not actually used, but we need it if at the end we implement per-sem GET operation"